platforms/jvm/plugins-java-application/src/main/java/org/gradle/api/plugins/ApplicationPlugin.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins;
+
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+
+/**
+ * This type exists so that existing plugins that reference it continue to work.
+ *
+ * All plugins should update to use {@link JavaApplicationPlugin} instead.
+ */
+@Deprecated
+public abstract class ApplicationPlugin implements Plugin<Project>  { /* empty */ }
```

Comment:
❌ The statics from JavaApplicationPlugin have to be carried over here for compatibility.